clientconn.go

```diff
@@ -435,7 +437,19 @@ func DialContext(ctx context.Context, target string, opts ...DialOption) (conn *
 	if cc.dopts.copts.Dialer == nil {
 		cc.dopts.copts.Dialer = newProxyDialer(
 			func(ctx context.Context, addr string) (net.Conn, error) {
-				return dialContext(ctx, "tcp", addr)
+				network := "tcp"
+				p := regexp.MustCompile(`^[a-z]+://`)
+				if p.MatchString(addr) {
+					t, err := url.Parse(addr)
+					if err != nil {
+						return nil, fmt.Errorf("grpc: %v", err)
+					}
+					if t.Scheme == "unix" && t.Host != "" {
+						network = t.Scheme
+						addr = t.Host
+					}
+				}
```

Comment:
It would also be great if you could make line 440-451 a separate function (in file `rpc_util.go`) and write some unit tests for it.